### PR TITLE
[New Conversation] Help Drawer

### LINKED ISSUE #5453: [New conversation] Create the help drawer
- Scoping [here](https://github.com/dust-tt/decisions/issues/198)
- no shortcuts in this card, done in a follow-up cards

Depends on [5446](https://github.com/dust-tt/dust/issues/5446)

///

As mentioned in the framing [issue](https://github.com/dust-tt/decisions/issues/212)

Full Figma [here](https://www.figma.com/design/QOxHwppG64GtPocpDhS6Y7/Product?node-id=5713-48822&m=dev)

### PR DESCRIPTION
**Pull request by philipperolet**
## Description

Fixes #5453

Help drawer for builders:
image

Help drawer for members:
image

Note: for the purposes of the drawer, this PR also provides an option to remove the attachment button for the inputbar

As per IRL with @Duncid :
- Leaving icons gray (vs green in figma) for now)
- when asking a question, mention @help is not required and will be added if not present 
- using other mentions in that box is an edge case, just letting the other mention run in addition of the @help one is therefore a viable solution

## Risk
None

## Deploy Plan
Front

### CODE REVIEW

File: front/pages/w/[wId]/assistant/new.tsx [hunk 1]
```diff
@@ -82,6 +83,7 @@ export default function AssistantNew({
   const [planLimitReached, setPlanLimitReached] = useState<boolean>(false);
   const sendNotification = useContext(SendNotificationsContext);
 
+  const [helpDrawer, setHelpDrawer] = useState<boolean>(false);
```
Comment (flvndvd) [inline front/pages/w/[wId]/assistant/new.tsx:86]:
```suggestion
  const [isHelpDrawerOpened, setIsHelpDrawerOpened] = useState<boolean>(false);
```

File: front/components/assistant/HelpDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        });
+        if (conversationRes.isErr()) {
+          sendNotification({
+            title: conversationRes.error.title,
+            description: conversationRes.error.message,
+            type: "error",
+          });
+        } else {
+          // We start the push before creating the message to optimize for instantaneity as well.
+          void router.push(
```
Comment (flvndvd) [inline front/components/assistant/HelpDrawer.tsx:120]: So it serves as an intent to create a new conversation, but there is no logic at all happening in the drawer once the user submits its first message?

File: front/components/assistant/HelpDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+  const router = useRouter();
+  const sendNotification = useContext(SendNotificationsContext);
+
+  const { submit: handleHelpSubmit } = useSubmitFunction(
+    useCallback(
+      async (input: string, mentions: MentionType[]) => {
+        const inputWithHelp = input.includes("@help")
+          ? input
+          : `@help ${input.trimStart()}`;
+        const mentionsWithHelp = mentions.some(
```
Comment (flvndvd) [inline front/components/assistant/HelpDrawer.tsx:95]: Why do we implement all this logic whereas we could either put `@help` in a sticky mention + only suggest this one in the input bar?

File: front/components/assistant/HelpDrawer.tsx (hunk 2 above)
Comment (philipperolet) [reply front/components/assistant/HelpDrawer.tsx:120]: Correct

File: front/components/assistant/conversation/input_bar/InputBar.tsx [hunk 4]
```diff
@@ -335,6 +337,7 @@ export function AssistantInputBar({
 
               <InputBarContainer
                 hideQuickActions={hideQuickActions}
+                hideAttachment={!!hideAttachment}
```
[the remark this answers is not in the document]
Comment (philipperolet) [reply front/components/assistant/conversation/input_bar/InputBar.tsx:340]: Agreed, but for a follow-up PR

File: front/components/assistant/HelpDrawer.tsx (hunk 3 above)
Comment (philipperolet) [reply front/components/assistant/HelpDrawer.tsx:95]: I tried to use sticky mentions, with the constraint, as discussed with ed, that we want there to be no mention in the input initially and that submitting a message without the mention should work (this is the help, user might not be familiar with mentions)

I could not make it work

File: front/components/assistant/conversation/input_bar/InputBar.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/components/assistant/conversation/input_bar/InputBar.tsx:340]: Haha, I'm not quite sure why we always defer implementing things properly at first 🤔

[the remark this answers is not in the document]
Comment (philipperolet) [reply front/components/assistant/conversation/input_bar/InputBar.tsx:340]: I think separating functionality PRs and refactoring PRs make the PRs easier to understand and the optional error investigations faster

File: front/components/assistant/HelpDrawer.tsx (hunk 3 above)
Comment (flvndvd) [reply front/components/assistant/HelpDrawer.tsx:95]: I'm not entirely convinced about this but as I wasn’t part of your discussion with @Duncid. IMHO, if we’ve agreed to proceed with mentions, we should avoid making exceptions. Exceptions can make it difficult for users to understand how to use our product. In this case, I believe that using a sticky mention for `@help` would not only serve the intended purpose but also help to educate our users about using mentions!

File: front/pages/w/[wId]/assistant/new.tsx [hunk 5]
```diff
@@ -83,7 +83,7 @@ export default function AssistantNew({
   const [planLimitReached, setPlanLimitReached] = useState<boolean>(false);
   const sendNotification = useContext(SendNotificationsContext);
 
-  const [helpDrawer, setHelpDrawer] = useState<boolean>(false);
+  const [showHelpDrawer, setShowHelpDrawer] = useState<boolean>(false);
```
Comment (flvndvd) [inline front/pages/w/[wId]/assistant/new.tsx:86]: We tend to use `isXXX` to reflect boolean states, don't we 😛

**Review by flvndvd — APPROVED** [review]

File: front/pages/w/[wId]/assistant/new.tsx (hunk 5 above)
Comment (philipperolet) [reply front/pages/w/[wId]/assistant/new.tsx:86]: True. Will update

File: front/components/assistant/HelpDrawer.tsx (hunk 3 above)
Comment (philipperolet) [reply front/components/assistant/HelpDrawer.tsx:95]: Discussed IRL, keeping as is

**Pull request merged**